### ISSUE TITLE
Style/NestedTernaryOperator should be able to `AllowElseIf: true`

### ISSUE THREAD

**Issue by nevans** [rubocop/rubocop#11008]
## Is your feature request related to a problem? Please describe.

I believe that nested ternary expressions are relatively familiar to most teams with Javascript/Typescript developers.  In general, I agree that ruby has superior readability by simply treating `if`/`elsif`/`else` as expressions with values.

But there are times when I think the nested ternary is succinct and to-the-point in a way that works very well for my team and probably others.  Specifically, when the negative clause contains another ternary expression, that is analogous to `if`/`elsif`/`else`.  We usually don't interpret `if`/`elsif`/`else` as "nested", although it logically is.  The same can be said for `cond1? ? result1 : cond2? ? result2 : default`.

When your language doesn't treat `if` as a value expression, this isn't so bad:
```javascript
// exploded.  This is how prettier handles long ternary expressions.
// But this style requires "\" line continuations in ruby's grammar.
const val = cond1
  ? result1
  : cond2
  ? result2
  : cond3
  ? result3
  : nomatch

// eslint can allow more succinctly combining conditions with their positive clause:
const val =
  cond1 ? result1 :
  cond2 ? result2 :
  cond3 ? result3 :
  nomatch

// both allow very succinct nested ternary expressions to remain on one line
const val = cond1 ? result1 : cond2 ? result2 : cond3 ? result3 : nomatch
```

And this is how that might translate into ruby:
```ruby
# The standard ruby approach: just use an `if/`elsif`/`else`.
# This is ideal when the conditions and/or the clauses are longer.
if cond1?
  result1
elsif cond2?
  result2
elsif cond3?
  result3
else
  default
end

# If all conditions and their results fit on a single line, that can
# feel too tall and expanded.  So I'll often write this:
if    cond1? then result1
elsif cond2? then result2
elsif cond3? then result3
else              default
end

# In the most succinct expressions, even that feels verbose.
# I'm usually okay with single-line nested ternary expressions, e.g:
cond1? ? result1 : cond2? ? result2 : cond3? ? result3 : default
```

## Describe the solution you'd like

I would like the following boolean configuration option:
```yaml
Style/NestedTernaryOperator:
  AllowElseIf: true
```

This would allow the negative clause of a ternary expression to be another ternary expression.  It would not allow nesting via a larger expression containing an embedded ternary.

```ruby
# bad
cond1? ? (cond2? ? (cond3? ? default : not3) : not2) : not1
cond1? ? (cond2? ? result2 : (cond3? ? result3 : default)) : not1
cond1? ? (cond2? ? result2 : not2) : (cond3? ? result3 : not3)

# good
cond1? ? result1 : cond2? ? result2 : cond3? ? result3 : default

# good, unless MultilineTernaryOperator is enabled
cond1? ? result1 :
  cond2? ? result2 :
  cond3? ? result3 :
  default
```

It would safely autocorrect ternary expressions like so:
```ruby
predicate ? (nested_predicate ? nested_positive : nested_negative) : negative
# should be converted into the following
!predicate ? negative : nested_predicate ? nested_positive : nested_negative
```

This interacts with `Style/NegatedIfElseCondition`: a negated condition should be tolerated when it pushes nesting into an "else if" clause.

## Describe alternatives you've considered

The guard clause and `if`/`elsif`/`else` variations, I listed above.

Alternately, if more customization is desired, `AllowNestedClause` could be added with the following values:
 * `never` (default): nesting is never allowed
 * `else`: the negative branch may be nested
 * `consistent`: one branch or the other may be nested, but not both, for an entire nested ternary expression
 * `either`: one branch or the other may be nested, but not both, per each individual ternary operator

```ruby
# AllowedClause: else/consistent/either => good; never => bad
cond1? ? result1 : cond2? ? result2 : cond3? ? result3 : default

# AllowedClause: consistent/either => good; never/else => bad
cond1? ? (cond2? ? (cond3? ? default : not3) : not2) : not1

# AllowedClause: either => good; never/else/consistent => bad
cond1? ? (cond2? ? result2 : (cond3? ? result3 : default)) : not1

# AllowedClause: never/else/consistent/either => bad
cond1? ? (cond2? ? result2 : not2) : (cond3? ? result3 : not3)
```

But `AllowElseIf` is simpler gives me everything want today.  I give the other option because I've read a lot of Typescript that would require `AllowNestedClause: consistent` or `either`.  Still, I think we should discourage `either`, and even supporting `consistent` seems unimportant to me.

## Additional context

Tangentially, I'm okay with certain very specific forms of multiline ternary expressions.  But for this issue, let's assume that `Style/MultilineTernaryOperator` and various `Layout` rules will cooperate if any special treatment for nested ternary operators is desired.

Here's the real code that prompted this ticket:
```ruby
    def auth_display_str
      oauth2? ? (!needs_reauth? ? "OAuth2" : "REAUTH") : "passwd"
    end
```

rubocop autocorrected it to the following, which feels slightly worse than the original to me:
```ruby
    def auth_display_str
      if oauth2?
        needs_reauth? ? "REAUTH" : "OAuth2"
      else 
        "passwd"
      end
    end
```

As you can see, it's succinct and not very complex, but the parentheses are an indication that we might want to simplify it further.  This resulted from old habits of mine: I first learned, decades ago, to put the common clauses first and handle exceptions in the "else".  I also liked highlighting the conditions by putting them all up-front.  But those heuristics tend towards deep ugly nesting of the "then" clause, with simple "else" clauses separated a long distance from their condition.  So I've become convinced that it's usually best to handle the simple cases first and push further conditions into the next `else if` clause.  Still, old habits die hard.

As such, I wanted to see it autocorrected to this:
```ruby
    def auth_display_str
      !oauth2? ? "passwd" : needs_reauth? ? "REAUTH" : "OAuth2"
    end
```

Or even a guard clause version:
```ruby
    def auth_display_str
      return "passwd" unless oauth2?
      return "REAUTH" if needs_reauth?
      "OAuth2"
    end
```

I also would have been okay with an `if`/`elsif`/`else` version:
```ruby
    def auth_display_str
      if !oauth2?
        "passwd"
      elsif needs_reauth?
        "REAUTH"
      else
        "OAuth2"
      end
    end
```
